remote/src/main/java/com/spotify/metrics/remote/OkRemote.java

```diff
@@ -89,4 +91,17 @@ public void onResponse(Call call, Response response) throws IOException {
         });
         return result;
     }
+
+    @Override
+    public void waitForAllCalls() {
+        Dispatcher dispatcher = client.dispatcher();
```

Comment:
I suggest adding a `client.dispatcher().executorService().shutdown();` as per http://square.github.io/okhttp/3.x/okhttp/ to shut down the executor service and thus disallow new calls.